src/lib/Frontend/JitFrontend.cpp

```diff
@@ -0,0 +1,67 @@
+#include "proteus/JitFrontend.hpp"
+
+#include "proteus/CompiledLibrary.hpp"
+#include "proteus/Hashing.hpp"
+
+#include <llvm/Bitcode/BitcodeWriter.h>
+#include <llvm/IR/Module.h>
+#include <llvm/IR/Verifier.h>
+#include <llvm/Support/MemoryBuffer.h>
+
+namespace proteus {
+
+JitModule::JitModule(const std::string &Target)
+    : Ctx{std::make_unique<LLVMContext>()},
+      Mod{std::make_unique<Module>("JitModule", *Ctx)},
+      TargetModel{parseTargetModel(Target)},
+      TargetTriple(::proteus::getTargetTriple(TargetModel)),
+      Dispatch(Dispatcher::getDispatcher(TargetModel)) {
+  Mod->setTargetTriple(TargetTriple);
+}
+
+void JitModule::compile(bool Verify) {
+  if (IsCompiled)
+    return;
+
+  if (Verify)
+    if (verifyModule(*Mod, &errs())) {
+      reportFatalError("Broken module found, JIT compilation aborted!");
+    }
+
+  SmallVector<char, 0> Buffer;
+  raw_svector_ostream OS(Buffer);
+  WriteBitcodeToFile(*Mod, OS);
+
+  // Create a unique module hash based on the bitcode and append to all
+  // function names to make them unique.
+  // TODO: Is this necessary?
+  ModuleHash =
+      std::make_unique<HashT>(hash(StringRef{Buffer.data(), Buffer.size()}));
+  for (auto &JitF : Functions) {
+    JitF->setName(JitF->getName() + "$" + ModuleHash->toString());
+  }
```

Comment:
Why is this done? To avoid name collisions with multiple specializations of a function?

Reply:
@ZwFink Let's keep the existing code for now and open an issue to investigate and refactor separately?